scanpipe/pipelines/fetch_scores.py

```diff
@@ -58,9 +59,26 @@ def check_scorecode_service_availability(self):
 
     def fetch_packages_scorecode_info(self):
         """Fetch ScoreCode information for each of the project's discovered packages."""
+        scorecard_policy = get_project_scorecard_thresholds(self.project)
+        worst_alert = None
+
         for package in self.project.discoveredpackages.all():
             if scorecard_data := ossf_scorecard.fetch_scorecard_info(package=package):
                 DiscoveredPackageScore.create_from_package_and_scorecard(
                     scorecard_data=scorecard_data,
                     package=package,
                 )
+
+                if scorecard_policy and scorecard_data.score is not None:
+                    try:
+                        score = float(scorecard_data.score)
+                        alert = scorecard_policy.get_alert_for_score(score)
+                    except Exception:
+                        alert = "error"
+
+                    order = {"ok": 0, "warning": 1, "error": 2}
+                    if worst_alert is None or order[alert] > order.get(worst_alert, -1):
+                        worst_alert = alert
+
+        if worst_alert is not None:
+            self.project.update_extra_data({"scorecard_compliance_alert": worst_alert})
```

Comment:
All this new logic should be moved to a pipe function. We want to keep the pipeline method a simple as possible. This would also make it easier to test (make sure to add one).